Vp.FSharp.Sql/TransactionScope.fs

```diff
@@ -2,83 +2,215 @@
 module Vp.FSharp.Sql.TransactionScope
 
 open System
+open System.Data.Common
 open System.Transactions
 
+open Vp.FSharp.Sql.Helpers
+
 
 [<Literal>]
 let DefaultIsolationLevel = IsolationLevel.ReadCommitted
 
 [<Literal>]
-let DefaultScopeOption = TransactionScopeOption.RequiresNew
+let DefaultScopeOption = TransactionScopeOption.Required
 
 [<Literal>]
 let DefaultTimeoutInSeconds = 30.
 
 let defaultTimeout = TimeSpan.FromSeconds(DefaultTimeoutInSeconds)
 
-let complete isolationLevel timeout scopeOption transaction =
+let private newTransactionScope isolationLevel timeout scopeOption =
+    let mutable transactionOptions = TransactionOptions()
+    transactionOptions.Timeout <- timeout
+    transactionOptions.IsolationLevel <- isolationLevel
+    new TransactionScope(scopeOption, transactionOptions, TransactionScopeAsyncFlowOption.Enabled)
+
+let private execute isolationLevel timeout scopeOption
+    (connection: #DbConnection) body =
     async {
-        let mutable transactionOptions = TransactionOptions()
-        transactionOptions.Timeout <- timeout
-        transactionOptions.IsolationLevel <- isolationLevel
-        use transactionScope = new TransactionScope(scopeOption, transactionOptions, TransactionScopeAsyncFlowOption.Enabled)
-        let! operationsResult = transaction
+        use transactionScope = newTransactionScope isolationLevel timeout scopeOption
+        connection.EnlistCurrentTransaction()
+        let! applyOutcome = body connection
+        return (transactionScope, applyOutcome)
```

Comment:
I don't get how this should work if `transactionScope` is disposed before being returned.

Reply:
Cause every now and then everybody makes mistakes, copy pasta and eventually trip over.

Is it a good enough explanation?